File: code/fib.py
# Fibonacci numbers, imperative style (Python)
def fibonacci(iterations):
    the_sum, first, second = 0, 0, 1  # initial seed values
    for i in range(iterations - 1):  # Perform the operation iterations - 1 times.
        the_sum = first + second
        first = second
        second = the_sum # Assign all the new values.
    return first  # Return the value when done.

File: code/fib.hs
-- Fibonacci numbers, functional style (Haskell)

-- describe an infinite list based on the recurrence relation for Fibonacci numbers
fibRecur first second = first : fibRecur second (first + second)
 
-- describe fibonacci list as fibRecurrence with initial values 0 and 1
fibonacci = fibRecurrence 0 1
 
-- describe action to print the 10th element of the fibonacci list
print (fibonacci !! 10)

